pkg/yang/yang.go

```diff
@@ -421,6 +423,7 @@ type LeafList struct {
 	Extensions []*Statement `yang:"Ext"`
 
 	Config      *Value   `yang:"config"`
+	Default     *Value   `yang:"default"`
```

Comment:
Should this be an array since https://tools.ietf.org/html/rfc7950#section-7.7.2 says there could be multiple of these statements? `default` has cardinality 0..n. Admittedly `Deviate` also got this wrong, though is probably out of scope for this change.

Reply:
good find @wenovus, I will change the code accordingly

Reply:
I think if you just change `*Value` to `[]*Value` it might work. I think ast.go might be using `append()` to populate slice values.

Reply:
@wenovus but how should I handle the `e.Default` assignment here - https://github.com/openconfig/goyang/pull/139/files#diff-c86a5ef7d87004179b80f56810efd550R568

entry's Default is a string; should I range over s.Default and create a `[]string` from Names of default statements which I later `strings.Join` to `e.Default`?

Reply:
Good point. I'm not sure which way to go for this issue. Your suggestion is straightforward; however, I don't see what delimiter to use to extract the individual values if we use `strings.Join` to create the value.

I see two options that may be more usable:
1. Add `Defaults []string` field for `Entry`, populated when it's a leaf-list.
 + Easier implementation.
 - Users of goyang will need to select the correct one.
2. Change the `Default` field to a slice type.
 + Single field for defaults. Users will simply use the `Entry.DefaultValue()` [method](https://github.com/openconfig/goyang/blob/633eccb6fa970a22464f8f9b22f23b26ff2487ed/pkg/yang/entry.go#L1441) to extract the singleton value.
 - Breaking change.

I'm kind of leaning towards 1. @robshakir @andaru would be happy to hear your thoughts.

Reply:
Could the `Defaults []string` be added to ListAttr?  From my reading of rfc7950, it's only in leaf-list that multiple defaults are allowed for a statement so might as well restrict this multi-default to that structure?

Reply:
Thanks @wingeng that sounds clever. However, that means the default value would be spread into two places.

@robshakir and I discussed, and decided to use option 2 coupled with the `SingleDefaultValue()` accessor. This way there is minimum chance of misuse for a new user.

Reply:
@wenovus thanks for taking this offline. Do I understand correctly that someone from OC will change this PR to align it with the chosen method? Or should I take a shot at it?

Reply:
I will get to this eventually. However you're more than welcome to take a stab and we'd be willing to review.

Reply:
Is there an example of that design approach that I can look at?

Reply:
Ok, I've implemented the option agreed on above.

One concern I have is that `SingleDefaultValue()` would return an empty string even if there was no default value. Since the default value *could* be the empty string, for example, we would not be supporting that.

As a result, I have added `HasDefault()` and `DefaultValues()` to help with checking for the existence of default values, and with getting all default values for a leaf-list type, respectively. `SingleDefaultValue()` would be useful for getting the default value for a non-leaf-list leaf entry.

Reply:
`HasDefault()` has been removed. See updated OP.